client/dashboard/sites/index.tsx

```diff
@@ -50,19 +50,27 @@ const DEFAULT_FIELDS: Field< Site >[] = [
 		label: __( 'Site' ),
 		enableGlobalSearch: true,
 		getValue: ( { item } ) => item.name || new URL( item.URL ).hostname,
+		render: ( { field, item } ) => (
+			<span style={ { overflowX: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap' } }>
```

Comment:
I noticed [chakra-ui](https://chakra-ui.com/docs/components/text#truncation) supports the `truncate` feature. It would be great if our `Text` component also supports it 🥲

Maybe it's acceptable to have a global classname like `span.truncate` to apply these styles 🤔

Reply:
We could make a `<Truncate>` component of our own, it would require a wrapping `span` but they're usually harmless. I like utility classes though. No need to reach for full blown components when an obvious lightweight solution exists.

But I seem to remember that this specific set of CSS rules doesn't work in every context. I can't remember the details, I feel like depending on whether you're in a `flex`, `position:relative`, or `inline-block` context sometimes you have to take a different approach? So I wouldn't want to refactor something out that doesn't actually work everywhere.

Or am I mis-remembering?

Reply:
chakra-ui and tailwindcss both use these 3 css rules 🤔

https://github.com/chakra-ui/chakra-ui/blob/a936404c54ebcb638569aedb9f84f7e540e371cb/packages/react/src/preset-base.ts#L1047

https://tailwindcss.com/docs/text-overflow

I guess it is good enough to use everywhere 🤷‍♂️